lib/BSON/DBRef.pm

```diff
@@ -130,12 +130,23 @@ sub TO_JSON {
     my $self = shift;
 
     if ( $ENV{BSON_EXTJSON} ) {
-        return {
-            '$ref' => $self->ref,
-            '$id'  => $self->id,
-            ( defined($self->db) ? ( '$db' => $self->db ) : () ),
-            %{ $self->extra },
-        };
+        my $id = $self->id;
+        $id = ref($id)
+            ? $id->TO_JSON
+            : { '$numberInt' => "$id" };
```

Comment:
I think this needs to do "looks_like_number", etc. and set this to be $numberInt, $numberLong, $numberDouble or else just put it in as an ordinary string.